package.json

```diff
@@ -178,5 +179,27 @@
     "hooks": {
       "pre-commit": "yarn run checkTs && lint-staged"
     }
+  },
+  "alias": {
+    "utils": "./src/utils",
+    "app": "./src/app",
+    "types": "./src/types",
+    "styles": "./src/styles",
+    "config": "./src/config",
+    "store": "./src/store",
+    "reportWebVitals": "./src/reportWebVitals.ts",
+    "vendors": "./src/vendors"
+  },
+  "targets": {
+    "web": {
+      "context": "browser",
+      "source": "src/index.html"
```

Comment:
I think Parcel has a bug, or `source` in `targets` doesn't do anything.
1. `yarn build` Done in **36.04s**
2. `yarn clean && npx parcel build --target web --dist-dir build` Same
3. `yarn clean && npx parcel build --target web --dist-dir build .` Same
4. `yarn clean && npx parcel build --target web --dist-dir build src/index.html` Built in **339ms**
5. `touch ~/yarn.lock` (create yarn.lock in one of the ancestor folders)
6. `yarn build` Takes a **really long time or errors**, I guess related to how much stuff is in ancestor folder
7. `yarn clean && npx parcel build --target web --dist-dir build` Same
8. `yarn clean && npx parcel build --target web --dist-dir build .` Same
9. `yarn clean && npx parcel build --target web --dist-dir build src/index.html` Built in **308ms**
10. `rm ~/yarn.lock`

The cause of 6., 7., 8. is wrong entries and wrong projectRoot in @parcel/core/lib/resolveOptions.js.

Do you get a large performance boost by specifying entry in command, too?

Reply:
Was playing with it a little bit and indeed there is an issue. If we care about use case where someone runs `touch ~/yarn.lock` then let's drop source in target and move it to command line. For performance boost I did not notice anything all builds take ~27s.